Zenventory/Services/CoreDataService.swift

```diff
@@ -9,41 +9,39 @@ import Foundation
 import Combine
 import CoreData
 
-class CoreDataService: ObservableObject, CoreDataManager {
-    let container: NSPersistentContainer
-    @Published var savedEntities: [ProductEntity] = []
-    var savedEntitiesPublisher: Published<[ProductEntity]>.Publisher { $savedEntities }
-    var savedEntitiesPublished: Published<[ProductEntity]> { _savedEntities }
+final internal class CoreDataService: ObservableObject, CoreDataManager {
+    @Published internal var savedEntities: [ProductEntity] = []
+
+    internal let container: NSPersistentContainer
+    internal var savedEntitiesPublisher: Published<[ProductEntity]>.Publisher { $savedEntities }
+    internal var savedEntitiesPublished: Published<[ProductEntity]> { _savedEntities }
 
     private var cancellables = Set<AnyCancellable>()
 
-    init() {
+    internal init() {
         container = NSPersistentContainer(name: "ProductsContainer")
-        container.loadPersistentStores { (description, error) in
-            if let error = error {
-                print("Error loading core data: \(error.localizedDescription)")
-            }
-        }
-        fetchProducts()
+        container.loadPersistentStores { (_, _) in }
+        try? fetchProducts()
```

Comment:
That's good that you've started to use `throws` functions. However in this case `try?` means that if `fetchProducts` function will catch any error, it'll be ignored. 

So if you want to handle this somehow then you can, for example, create some mechanism to show an alert on your `View` if there is an error. But it depends on what result you expect if there is any error.